### ISSUE TITLE
go client doesn't appear to pass along the session cookie

### ISSUE THREAD

**Issue by jess-sheneberger** [ory/kratos#1459]
This is regarding the go client in repo ory/kratos-client-go, v0.6.3-alpha.1.  That repo doesn't appear to have an issues section so I'm opening the issue here. Please let me know if there is a more appropriate place for this.

This code underwent some major changes since v0.5 and I discovered after upgrading that when I try to supply a session cookie to the client it doesn't appear to work. I might be doing it wrong, I'm trying to follow the instructions here: https://github.com/ory/kratos-client-go/tree/v0.6.3-alpha.1#sessioncookie

**Reproducing the bug**

I used a code snippet like:
```
configuration := kratos.NewConfiguration()
configuration.Servers[0].URL = "localhost"
kc := kratos.NewAPIClient(configuration)
auth = context.WithValue(auth, kratos.ContextAPIKeys, map[string]kratos.APIKey{
    "ory_kratos_session": kratos.APIKey{Key: "XYZ"},
})
op := kc.PublicApi.ToSession(auth)
session, response, err := op.Execute()
```

...and response contains a 401 error. Looking through the client code I don't see it using the APIKey or the associated collection anywhere, so I'm guessing it's simply not passing it along. 

**Expected behavior**

I expect a 200 response from the /session/whoami call. That's what I get when I supply a session token via `op.XSessionToken()`, or when I roll back to ory/kratos-client-go v0.5.4-alpha.1 and supply a session cookie via `WhoamiParams.WithCookie()`.

**Environment**
ory/kratos-client-go v0.6.3-alpha.1

**Comment by aeneasr** [comment]
Hello, sorry for the frustrating experience here. We're currently trying to figure out how to generate specs which allow us to have a consistent experience across all the different programming languages.

My suggestion is to use an SDK prior to 0.6.3 as those use the previous sessionWhoAmI method. We'll be streamlining the SDKs with the 0.7 milestone, see: https://github.com/ory/kratos/issues/1424

**Comment by aeneasr** [comment]
This is merged, 0.7 is being released today, closing! :) We now have also examples for golang: https://github.com/ory/kratos/tree/master/examples/go
